### DISCUSSION TITLE
Defining asset group dependencies in assets/modify group asset

### DISCUSSION THREAD

**Discussion by rytis-prisero** [dagster-io/dagster#18698] [Q&A] [answered]
Hello, is it possible to define dependency between asset groups.
E.g. I have a groups DBT_FACEBOOK (depends on AIRBYTE_FACEBOOK) and AIRBYTE_FACEBOOK(ingestion of data ).
Both sources are created dynamically so I would need to assign dependency only using group names. Thank you for your help.

Maybe there is a way to modify a group asset? Really struggling with the Dagster as the examples are very trivial.

**Comment by rytis-prisero** [comment]
Here is my current asset.py file.
```python
from dagster import AssetExecutionContext, asset
from dagster_dbt import DbtCliResource, dbt_assets
from dagster_airbyte import airbyte_resource, load_assets_from_airbyte_instance

from .constants import dbt_manifest_path

airbyte_instance = airbyte_resource.configured(
    {
        "host": "localhost",
        "port": "",
        "username": "airbyte",
        "password": {"env": "AIRBYTE_PASSWORD"},
        "use_https": True
    }
)

def connection_to_group(name: str) -> str:
    if name.startswith('FCB_'):
        return 'a_fcb'
    elif name.startswith('Google'):
        return 'a_gads'
    elif name.startswith('Adsense'):
        return 'a_adsense'
    elif name.startswith('Exchange'):
        return 'a_currency'
    else:
        return 'another_group'

airbyte_assets = load_assets_from_airbyte_instance(
    airbyte_instance,
    workspace_id="1cee28a1-b334-4e69-948f-cc3e9681d678",
    connection_to_group_fn=connection_to_group
)

@dbt_assets(manifest=dbt_manifest_path)
def prisify_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
    yield from dbt.cli(["build"], context=context).stream()```

**Comment by rytis-prisero** [comment]
Maybe there is another way how to manipulate dynamically created assets?

**Comment by rexledesma** [comment, marked as answer]
You cannot specify explicit dependencies between groups. You must specify dependencies at the asset level.

To do this, you should ensure that the asset keys of your Airbyte assets match up with the asset keys of your dbt sources.

References:
- [Defining upstream dependencies to dbt models](https://docs.dagster.io/integrations/dbt/reference#upstream-dependencies)

**Reply by rytis-prisero** [reply]
What if my airbyte assets do not match dbt sources. E.G. I have a DBT macro that scans table names (I have 200 facebook accounts) and then does a union for them. How could I define a particular DBT asset to be dependant on particular group of Airbyte assets? Maybe I can create other assets using the initial airbyte_assets?
Thanks @rexledesma for your initial answer!
